core/src/frame_lifecycle.rs

```diff
@@ -124,7 +124,7 @@ pub fn run_inner_goto_frame<'gc>(
     removed_frame_scripts: &[DisplayObject<'gc>],
     initial_clip: MovieClip<'gc>,
 ) {
-    if initial_clip.swf_version() <= 9 {
+    if initial_clip.swf_version() <= 9 && initial_clip.movie().is_action_script_3() {
```

Comment:
Is this line also only about improving AVM1 stubs? If so, how?

Reply:
This should fix the `Encountered stub: AVM2 flash.display.MovieClip.goto with SWF 9 movie` warning appearing on AVM1 movies that attempt to run a goto. The stub warning should only appear in AVM2 because it is only relevant to AVM2.

Reply:
Ah, I see, okay!